zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/InterpreterGroup.java

```diff
@@ -167,14 +159,7 @@ public void close(String sessionId) {
     LOGGER.info("Close interpreter group " + getId() + " for session: " + sessionId);
     List<Interpreter> intpForSession = this.get(sessionId);
     close(intpForSession);
-
-    if (remoteInterpreterProcess != null) {
-      remoteInterpreterProcess.dereference();
-      if (remoteInterpreterProcess.referenceCount() <= 0) {
```

Comment:
```
public void close(String sessionId) {
...
if (remoteInterpreterProcess.referenceCount() <= 0) {
```
This code reduces the reference count only for a specific session, and terminates the process when the reference count finally reaches zero.
However, if you look at the modified code, it seems that if you are asked to terminate a session, it will reduce all reference counts for all unrelated sessions and force the process to terminate. (call to same method - terminateRemoteProcess())
In my opinion, there seems to be some room for problems.